sub-packages/bionemo-moco/src/bionemo/moco/schedules/inference_time_schedules.py

```diff
@@ -458,3 +457,160 @@ def generate_schedule(
+
+        Approach benefits:
+        - Can improve inference performance
+        - Prevents oversampling of less-informative time steps
+        - Prevents undersampling of critical time windows
+        - Easily adapted into any model architecture leveraging flow-matching
+        - Sample-eficient way to generate data
+
+        Args:
+            predictor (Callable[[Tensor, Tensor], Tensor]): A callable (e.g., a function
+                or functools.partial) that takes a time tensor `t` and a data
+                tensor `x` and returns the predicted vector field `v`. 
+
+            x_0_sampler (Callable[[int], Tensor]): A function that takes a batch size
+                and returns a tensor of samples from the initial distribution p0.
+
+            x_1_sampler (Callable[[int], Tensor]): A function that takes a batch size
+                and returns a tensor of samples from the target distribution p1.
+
+            nsteps (int): The final number of time steps for the inference schedule.
+
+            n_schedule_points (int): The number of points used to approximate the
+                cumulative entropy curve. Higher is more accurate but slower.
+
+            batch_size (int): Batch size for calculating divergence at each time step.
+
+            inclusive_end (bool): If True, include 1.0, otherwise end just before.
+
+            min_t (Float): The minimum time value for the schedule.
+
+            direction (Union[TimeDirection, str]): 'UNIFIED' for forward (0->1) or
+                'DIFFUSION' for reverse (1->0).
+
+            device (Union[str, torch.device]): The device for computation.
+        """
+        super().__init__(nsteps, inclusive_end, min_t=min_t, direction=direction, device=device)
+        self.predictor = predictor
+        self.x_0_sampler = x_0_sampler
+        self.x_1_sampler = x_1_sampler
+        self.n_schedule_points = n_schedule_points
+        self.batch_size = batch_size
+
+    def hutchinson_divergence(model: torch.nn.Module, t: Tensor, x: Tensor) -> Tensor:
+        """
+        Estimates the divergence of a vector field defined by a model using Hutchinson's method.
+
+        Args:
+            model (torch.nn.Module): The neural network representing the vector field v(x, t).
+                                    It must accept a single tensor of shape [B, D+1]
+                                    where D is the dimension of x.
+            t (Tensor): A tensor of time values, shape [B, 1].
+            x (Tensor): A tensor of positions, shape [B, D].
+
+        Returns:
+            Tensor: The estimated divergence for each sample in the batch, shape [B].
+        """
+        # Detach and set requires_grad=True to compute gradients with respect to x
+        x = x.detach().requires_grad_(True)
+        
+        # Generate a random vector from the Rademacher distribution ({-1, 1})
+        epsilon = (torch.randint_like(x, 0, 2) * 2 - 1).to(x.dtype)
```

Comment:
add a generator if possible to be able to control the seed like in the other interpolants. This can likely be done in the init where maybe we can reuse the interpolants generator if not create a new one